crates/ln-dlc-node/src/ldk_node_wallet.rs

```diff
@@ -132,7 +142,17 @@ where
             }
         }
 
-        Ok(psbt.extract_tx())
+        let transaction = psbt.extract_tx();
+
+        let prev_outpoints = transaction
+            .input
+            .iter()
+            .map(|input| input.previous_output)
+            .collect::<Vec<_>>();
+
+        locked_out_points.extend(prev_outpoints);
```

Comment:
🔧 It might be a good idea to take this lock twice rather than holding it for an extended period of time. What do you think, @klochowicz?